src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -3558,33 +3559,42 @@ void InstrumentClipView::drawDrumName(Drum* drum, bool justPopUp) {
 			newText = buffer;
 
 			if (drum->type == DrumType::GATE) {
+
 				strcpy(buffer, "Gate channel ");
 				intToString(((GateDrum*)drum)->channel + 1, &buffer[13]);
 				indicator_leds::blinkLed(IndicatorLED::CV, 1, 1);
 			}
 			else { // MIDI
-				uint8_t drumNote = ((MIDIDrum*)drum)->note;
-				char channelString[10];
-				char noteString[5];
+				char topLine[30];
+				char noteLabel[5];
+
+				sprintf(topLine, "CH: %d  N#: %d", ((MIDIDrum*)drum)->channel + 1, ((MIDIDrum*)drum)->note);
+				noteCodeToString(((MIDIDrum*)drum)->note, noteLabel);
 
-				strcpy(buffer, "CHAN ");
-				intToString(((MIDIDrum*)drum)->channel + 1, channelString);
-				strcat(buffer, channelString);
-				strcat(buffer, ", NOTE ");
+				const char* lines[] = {topLine, noteLabel};
 
-				noteCodeToString(drumNote, noteString, nullptr);
-				strcat(buffer, noteString);
+				char bufferLines[50];
+				concatenateLines(lines, sizeof(lines) / sizeof(lines[0]), buffer);
 				indicator_leds::blinkLed(IndicatorLED::MIDI, 1, 1);
 			}
 		}
 
 		display->popupText(newText);
 	}
 	else {
+
 		char buffer[7];
 
 		if (!drum) {
 			newText = "NONE";
+
+basicDisplay:
+			if (justPopUp && currentUIMode != UI_MODE_AUDITIONING) {
```

Comment:
Can you make this a function instead of continuing to use a goto?

Reply:
This was pre-existing code, the order was slightly modified, I think? I'll take a look again, I just remember this was a side effect and when i looked closely at the diff i remember realizing it was moved not added.

Reply:
Ya but that's always a good opportunity for some cleanup!